### PR TITLE
Removed needless functions from an io.ascii test helper file #15782

### LINKED ISSUE #15782: Remove needless functions from an `io.ascii` test helper file
### What is the problem this feature will solve?

A helper file in `io.ascii` tests contains three functions that serve little purpose: https://github.com/astropy/astropy/blob/fefe2261fcf09751d052d80bd8143e72f18f6aea/astropy/io/ascii/tests/common.py#L27-L37

### Describe the desired outcome

The aforementioned functions should be removed and any tests still using them should be updated accordingly. Pull requests targeting one or more functions are welcome.

Potential contributors can have a look at #16644 that removed the helper functions from one file and use that as an example.

### Additional context

`assert_equal()` and `assert_true()` should be inlined, but `assert_almost_equal(a, b)` should be replaced with [`np.testing.assert_allclose(a, b)`](https://numpy.org/doc/stable/reference/generated/numpy.testing.assert_allclose.html#numpy.testing.assert_allclose) instead of `assert np.allclose(a, b)` because the dedicated testing function produces a better error message if the check fails.

### Performed removals

- [ ] `assert_equal()`
- [ ] `assert_almost_equal()`
- [ ] `assert_true()`

### PR DESCRIPTION
**Pull request by helloworld-chhanda**
### Description

This pull request is to address ...

Fixes #15782

Description:
`assert_equal` , `assert_true` has been removed from "common.py" as well as other files.
`assert_allclose` has been used instead of `assert_almost_equal`.

Performed removals
 - [ ] assert_equal()
 - [ ] assert_almost_equal()
 - [ ] assert_true()

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: astropy/io/ascii/tests/common.py [hunk 1]
```diff
@@ -25,13 +22,13 @@ def teardown_function(function):
 
 
 # Compatibility functions to convert from nose to pytest
-def assert_equal(a, b):
-    assert a == b
+# def assert_equal(a, b):
+#     assert a == b
```
Comment (eerovaher) [inline astropy/io/ascii/tests/common.py:26]: The functions should be deleted, not commented out.

File: astropy/io/ascii/tests/test_cds_header_from_readme.py [hunk 2]
```diff
@@ -34,22 +33,20 @@ def test_description():
...
+        assert table["Cluster"].description == "Cluster name"
+        assert table["Star"].description == ""
+        assert table["Wave"].description == "wave ? Wavelength in Angstroms"
+        assert table["El"].description == "a"
+        assert table["ion"].description == "- Ionization stage (1 for neutral element)"
+        assert (
+            table["loggf"].description ==
             "log10 of the gf value - logarithm base 10 of stat. weight times "
             "oscillator strength",
         )
```
Comment (eerovaher) [inline astropy/io/ascii/tests/test_cds_header_from_readme.py:46]: Ruff correctly detects an [F631 (assert-tuple)](https://docs.astral.sh/ruff/rules/assert-tuple/) violation here. [Our developer documentation recommends using the `pre-commit` tool](https://docs.astropy.org/en/latest/development/workflow/development_workflow.html#pre-commit), which would automatically detect such mistakes before a commit was even made.
```suggestion
        assert (
            table["loggf"].description ==
            "log10 of the gf value - logarithm base 10 of stat. weight times "
            "oscillator strength"
        )
```

File: astropy/io/ascii/tests/test_c_reader.py [hunk 3]
```diff
@@ -13,6 +13,7 @@
 import pytest
 from numpy import ma
 
+from numpy import testing as npt
```
Comment (eerovaher) [inline astropy/io/ascii/tests/test_c_reader.py:16]: I'm not sure if I would use this `npt` alias, but it is used in some of our tests already, so there is precedent for this.

**Review by eerovaher — CHANGES_REQUESTED** [review]
I had a quick look at the changes to `io.ascii` and it seems like only a couple of minor changes are needed there. However, a pull request addressing #15782 has no reason to edit anything outside `io.ascii`, so all the changes elsewhere need to be undone.

**Comment by pllim** [comment]
Maybe we should remove "package-novice" label from https://github.com/astropy/astropy/issues/15782

File: astropy/io/ascii/tests/test_c_reader.py (hunk 3 above)
Comment (dhomeier) [reply astropy/io/ascii/tests/test_c_reader.py:16]: –1 on adopting this. We have discussed this in one of the various PRs addressing the same issue before, and already concluded that `from numpy.testing import assert_allclose` etc. is the preferred style (and already used with only the aforementioned 1 or 2 exceptions).

**Comment by dhomeier** [comment]
Also should note this in #16266 or ping @stefanarseneau on a response to https://github.com/astropy/astropy/pull/16266#issuecomment-2052345182.

**Comment by pllim** [comment]
We have at least 4 PRs now trying to address that issue and the diff here is too messy. I think we should close this without merge.

* https://github.com/astropy/astropy/pull/15787
* https://github.com/astropy/astropy/pull/15839
* https://github.com/astropy/astropy/pull/16266
* https://github.com/astropy/astropy/pull/16383 (this PR)

**Comment by eerovaher** [comment]
Out of all the related pull requests this seems to be closest to completion. The diff is not overly messy. It is far too large because of all the changes outside `io.ascii`, but undoing those changes is very simple and does not require review.

**Comment by eerovaher** [comment]
To undo all the changes outside `io.ascii` in your local repository the following might be useful:
```shell
$ git reset --hard $(git merge-base main @)
$ git restore --source=@{u} astropy/io/ascii
```

**Comment by pllim** [comment]
Superseded by https://github.com/astropy/astropy/pull/16644 . Thanks anyway!

**Pull request closed without merging**